src/components/backend-ai-login.ts

```diff
@@ -606,6 +606,16 @@ export default class BackendAILogin extends BackendAIPage {
     this.api_endpoint = this.api_endpoint.trim();
     if (this.connection_mode === 'SESSION') {
       // this.block(_text('login.PleaseWait'), _text('login.ConnectingToCluster'));
+      if (!window.location.href.startsWith(this.api_endpoint)) {
```

Comment:
@adrysn Do we really need this part? Since every login logic uses check_login prior to login, this code do the same behavior along with the logic in check_login.

Reply:
Ah, we can switch between API and session mode. Nevermind;

Reply:
Let's make it as function, not a duplicated code blocks.